### ISSUE TITLE
[BUG] Adding an immutable record to LiteDB gives an NRE

### ISSUE THREAD

**Issue by isaacabraham** [litedb-org/LiteDB#1541]
**Version**: LiteDB 5.0.3, Windows, NetCore3.1

**Describe the bug**
Adding an immutable record to LiteDB gives an NRE, although the document is actually inserted.

**Code to Reproduce**
```fsharp
open LiteDB

type Customer = {
    Id : int
    Name : string
    Phones : string array
    IsActive : bool
}

let db = new LiteDatabase("MyData.db")
let col = db.GetCollection "customers"

let customer =
    { Id = 0
      Name = "John Doe"
      Phones = [| "8000-0000"; "9000-0000" |]
      IsActive = true }

// Insert new customer document (Id will be auto-incremented)
col.Insert customer |> ignore
```

**Expected behavior**
The record should be inserted without an exception.

`System.NullReferenceException: Object reference not set to an instance of an object.
   at LiteDB.LiteCollection`1.Insert(T entity)`

**Additional context**
You can work around this by marking the record type with `[<CLIMutable>]`. However, this is non-idomatic from an F# perspective.

**Comment by nightroman** [comment]
FWIW https://github.com/Zaid-Ajaj/LiteDB.FSharp#id-auto-incremented

**Comment by isaacabraham** [comment]
Yes, maybe that's all that needs to be done. Note that LiteDB.FSharp is not compatible with v5 of LiteDB cc @Zaid-Ajaj

**Comment by Zaid-Ajaj** [comment]
There is already an [opened issue](https://github.com/Zaid-Ajaj/LiteDB.FSharp/issues/45) about updating LiteDB.FSharp to latest LiteDB version. I won't be able to work on it any time soon due to other prioritized projects. Feel free to send PRs for the issue in subject, I will be happy to review them

**Comment by humhei** [comment]
> You can work around this by marking the record type with `[<CLIMutable>]`. However, this is non-idomatic from an F# perspective.

Another workaround is to change `int` to `ObjectId` without adding `[<CLIMutable>]` attribute
```fsharp
type Customer = {
    Id : ObjectId
    Name : string
    Phones : string array
    IsActive : bool
}
```

**Comment by humhei** [comment]
A issue is the record type using `ObjectId` is that will not support `comparision`
![image](https://user-images.githubusercontent.com/25994449/81244356-c83f7800-9044-11ea-999e-eb790a273f09.png)

**Comment by lbnascimento** [comment]
@humhei This happens because `ObjectId` implements `IComparable<ObjectId>`, but not `IComparable` (an oversight, I believe). Try using `Guid` as Id, it should work.

**Comment by JKamsker** [comment]
Consolidated into #2811, which combines this report with the related ones and tracks the fix. Closing this as a duplicate — please follow #2811, and comment there if your case is different.
